Source/Order.swift

```diff
@@ -27,24 +27,59 @@ import Foundation
 /**
 Represents an order placed by the user to be tracked using `ButtonMerchant.trackOrder(order)`.
  */
-final public class Order: NSObject, Codable {
+@objcMembers final public class Order: NSObject, Codable {
 
     /**
      The order identifier (required).
      */
-    let id: String
+    public var id: String
 
     /**
      The total order value in pennies (e.g. 3999 for $39.99)
-     or the smallest decimal unit of the currency.
+     or the smallest decimal unit of the currency. (default is 0)
      */
-    private(set) var amount: Int64
+    @available(*, deprecated)
+    private(set) var amount: Int64 = 0
 
     /**
      The ISO 4217 currency code (default is USD).
      */
-    private(set) var currencyCode: String
+    public var currencyCode: String
 
+    /**
+     The purchase date for the order.
+     */
+    public var purchaseDate: Date?
+
+    /**
+     The customer-facing order id
+     */
+    public var customerOrderId: String?
+
+    /**
+     A list of the line item details that comprise the order
+     */
+    public var lineItems: [LineItem]?
```

Comment:
The required params passed on init() should not be able to be changed after creating the Order object.